frontend/shared/GithubAuthButton.test.jsx

```diff
@@ -0,0 +1,207 @@
+import React from 'react';
+import { render, screen, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { spy, stub, restore } from 'sinon';
+import '@testing-library/jest-dom';
+import globals from '../globals';
+
+import api from '@util/federalistApi';
+import GithubAuthButton from './GithubAuthButton';
+
+const BUTTON_TEXT = `Connect with GitHub`;
+const text = 'Hello World';
+
+// We need to create the MessageEvent to set the origin
+// and data when testing
+// https://github.com/jsdom/jsdom/issues/2745#issuecomment-1207414024
+function createPostMessageEvent(data, origin) {
+  return new MessageEvent('message', {
+    source: window,
+    origin,
+    data,
+  });
+}
+
+function createStubs() {
+  const onFailure = spy();
+  const onSuccess = spy();
+  const apiStub = stub(api, 'revokeApplicationGrant').resolves();
+  const focusStub = stub().resolves();
+  const closeStub = stub().returns();
+  const windowStub = stub(window, 'open').returns({
+    focus: focusStub,
+    close: closeStub,
+  });
+
+  return {
+    onFailure,
+    onSuccess,
+    apiStub,
+    focusStub,
+    closeStub,
+    windowStub,
+  };
+}
+
+describe('<GithubAuthButton/>', () => {
+  beforeAll(() => {
+    // Set window origin based on globals APP_HOSTNAME
+    window['origin'] = globals.APP_HOSTNAME;
+
+    // Set screen height and width
+    window['screen'] = {
+      width: 1200,
+      height: 1000,
+    };
+  });
+
+  afterEach(() => restore());
```

Comment:
Since the stubs are getting reset after each test, we could probably create them inside a `beforeEach` and avoid redefining for each test. We can also create the user here and scope it to the describe block (rather than each test individually)

Reply:
Makes sense to me.